src/librustc_typeck/check/intrinsic.rs

```diff
@@ -319,6 +322,27 @@ pub fn check_intrinsic_type<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
                 (0, vec![ptr_ty, tcx.types.usize], tcx.types.usize)
             },
 
+            "va_start" | "va_end" => {
+                match va_list_ty {
+                    Some(va_list_ty) => (0, vec![va_list_ty], tcx.mk_nil()),
+                    None => bug!("va_list lang_item must be defined to use va_list intrinsics")
+                }
+            }
+
+            "va_copy" => {
+                match va_list_ty {
+                    Some(va_list_ty) => (0, vec![va_list_ty, va_list_ty], tcx.mk_nil()),
```

Comment:
Although, you might want to make a closure for `va_list_ty` instead, since these two arguments want separate lifetimes (which would mean that instead of `ty::BrAnon(0)` in my snippet above, you'd have `ty::BrAnon(i)` with `i` being `0` vs `1` in this case - in a similar fashion to `param(0)`; if you wanted to you could also go the way of adding a `lifetime` closure like `param`).

Reply:
Good point, I'll do that.